### ISSUE TITLE
Bundle Python Interpreter

### ISSUE THREAD

**Issue by Turakar** [astral-sh/uv#7865]
For system wide installations of applications build with `uv`, it would be nice to have an option to build an execution environment which is self contained and does not rely on the user's local files (`~/.local/share/uv`). My usecase for this would be an application that I want to install to a server for system-wide usage. At the moment, you can achieve this with the following commands (with some path adjustments):

```bash
uv venv --no-cache
uv pip install --no-cache -r requirements.lock
cp -r ~/.local/share/uv/python/cpython-3.12.5-linux-x86_64-gnu python
rm .venv/bin/python
ln -s python/bin/python3.12 .venv/bin/python

# To run the program
.venv/bin/python <your script.py>
```

I.e., you can replace the symlinked python interpreter by a bundled copy.

I am not entirely sure how this should be implemented. In a first simple step, one could add a `--bundle-python` flag to `uv venv` which copies the downloaded Python interpreter to, e.g., `.venv/python` and uses it instead of a path in `~/.local/share/uv`.

However, you can take this idea further, up to a `uv install` command which would do something similar to `make install`.

**Comment by Turakar** [comment]
Ok, you can kinda do this already by passing `UV_PYTHON_INSTALL_DIR` [(docs)](https://docs.astral.sh/uv/configuration/environment/) to `uv venv`.

**Comment by zanieb** [comment]
I think we want to add this feature.

**Comment by zanieb** [comment]
Related

- https://github.com/astral-sh/uv/issues/6970

**Comment by danielgafni** [comment]
Ohh til `UV_PYTHON_INSTALL_DIR` exists. 

I was looking for this parameter in CLI `--help` and it's not mentioned there. 

Perhaps 
1. The env var should be mentioned in `uv python *` docs
2. `uv python install *` commands should take an `--install-dir` argument?

**Comment by zanieb** [comment]
Yes we should mention it in the help documentation. I'd also be down to accept a target directory in the CLI.

**Comment by danielgafni** [comment]
Ok, I started working on this locally. Does `--install-dir` name work?

**Comment by kevinmarks-b** [comment]
I'm trying to do this to get the Python binary into an RPM for deployment on AWS. 
However, it seems impossible to get uv to see the python installed in a nearby directory with `UV_PYTHON_INSTALL_DIR` - even if I do as the OP did and symlink it to the install, it destroys that version and reinstalls when I use `uv run` to the ~/.local/share/uv/python/ one instead. 
Now I'm trying to do this in a make file, so `UV_PYTHON_INSTALL_DIR` may not persist into the runtime. Is there a way to make uv find it?
